test/pools.jl

```diff
@@ -0,0 +1,99 @@
+using ConcurrentUtilities, Test
+
+@testset "Pools" begin
+    pool = Pool(Int; max=3)
+    # acquire an object from the pool
+    x1 = acquire(() -> 1, pool)
+    # no existing objects in the pool, so our function was called to create a new one
+    @test x1 == 1
+    # release back to the pool for reuse
+    release(pool, x1)
+    # acquire another object from the pool
+    x1 = acquire(() -> 2, pool)
+    # this time, the pool had an existing object, so our function was not called
+    @test x1 == 1
+    # but now there are no objects to reuse again, so the next acquire will call our function
+    x2 = acquire(() -> 2, pool)
+    @test x2 == 2
+    x3 = acquire(() -> 3, pool)
+    @test x3 == 3
+    # the pool is now at capacity, so the next acquire will block until an object is released
+    tsk = @async acquire(() -> 4, pool; forcenew=true)
+    yield()
+    @test !istaskdone(tsk)
+    # release an object back to the pool
+    release(pool, x1)
+    # now the acquire can complete
+    x1 = fetch(tsk)
+    # even though we released 1 for reuse, we passed forcenew, so our function was called to create new
+    @test x1 == 4
+    # error to try and provide a key to a non-keyed pool
+    @test_throws ArgumentError acquire(() -> 1, pool, 1)
```

Comment:
I tried stepping through these tests locally, and i found the amount of state quite inconvenient

it made it hard to follow which tests depended on what came before and which didn't

can we introduce `@testset`s to make these things clearer, and create a fresh `Pool` for each testset e.g. this could just be
```suggestion
    @testset "keyed acquire on non-keyed pool" begin
        # error to try and provide a key to a non-keyed pool
        pool = Pool(Int, max=2)
        @test_throws ArgumentError acquire(() -> 1, pool, 1)
    end
```
```